
Fix: uClibc + Fuse Deadlock problem!

 http://www.mail-archive.com/uclibc@uclibc.org/msg01074.html

Kasia Binam
Tue, 18 Dec 2007 13:12:22 -0800

I posted a similar issue a while back for daemon() and pthreads in 
0.29.9, but got no response -- probably because I posted TMI :)   I'm 
not sure this is the correct way to fix this, but the libc_hidden_proto 
has daemon() use the libc fork instead of libpthread's fork.  If I just 
take it out, threads do not hang after daemon() as before.

diff -urbB uClibc-0.9.29/libc/unistd/daemon.c uClibc/libc/unistd/daemon.c
--- uClibc-0.9.29/libc/unistd/daemon.c	2006-07-05 23:52:39.000000000 +0400
+++ uClibc/libc/unistd/daemon.c	2008-11-24 14:19:57.469894000 +0300
@@ -54,7 +54,7 @@
 libc_hidden_proto(dup2)
 libc_hidden_proto(setsid)
 libc_hidden_proto(chdir)
-libc_hidden_proto(fork)
+/* libc_hidden_proto(fork) */
 
 int daemon( int nochdir, int noclose )
 {
